### DISCUSSION TITLE
How to set up connection with Microsoft Entra ID

### DISCUSSION THREAD

**Discussion by tfitzmac** [microsoft/promptflow#3600] [General]
I am trying to add a connection to an Azure OpenAI resource using my Azure credentials instead of an API key. I see the [instructions for adding a connection with Entra ID](https://microsoft.github.io/promptflow/how-to-guides/manage-connections.html#authenticate-with-microsoft-entra-id). I have added auth_mode: "meid_token" to the file for creating my connection. However, when I try to use the connection, I get an error that states: "Please pass one of `api_key`, `azure_ad_token`, `azure_ad_token_provider`, or the `AZURE_OPENAI_API_KEY` or `AZURE_OPENAI_AD_TOKEN` environment variables"

I am logged in with az login and I have the latest version azure-identity installed.

What values am I missing for creating the connection?

**Comment by brynn-code** [comment]
Hi, thank you for reaching us, below is the required promptflow and promptflow-tools package version, could you please check them at first?
![image](https://github.com/user-attachments/assets/f2872fbf-d521-4337-ae73-09f73f7d058c)

If your environment meets the requirements, then please let us know the tool you are using for future investigation, it would be better if we could see the **flow dag yaml** file and the **connection yaml** file.

**Reply by tfitzmac** [reply]
Thank you for your help. That fixed my issue. I was assuming that when I upgraded promptflow (currently using 1.14), prompflow-tools was also being upgraded. But, I see that they were out-of-sync. When I upgraded promptflow-tools to 1.4, it worked.

**Comment by geciciserkan** [comment]
@brynn-code another question related with this, 

We connect to Azure AI Search by below config which use API KEY in secret section. How can we update this config to use managed idendity for connection instead of API KEY.  Please note that we had achived to use managed idendity for Azure OpenAI connection by above configuration (auth_mode)

$schema: https://azuremlschemas.azureedge.net/promptflow/latest/CustomConnection.schema.json
name: custom_search_connection
type: custom
configs:
  endpoint: https://########.search.windows.net
  index_name: ####
secrets:
  key: ######
